### PR TITLE
feat(editor): Overhaul document title management

### LINKED ISSUE #8861: Page title not always updated
### Bug Description

Starting from a n8n page at a individual workflow, the page title is not updated if you click on left menubar at:

- /workflows
- /credentials
- /variables
- /settings

It works with /executions 

### To Reproduce

1. Open an individual workflow
2. Click on /workflows in the left menu bar
3. Check page title in browser

### Expected behavior

By clicking on

- /workflows
- /credentials
- /variables
- /settings

page title should be updated accordingly (e.g. n8n - workflows)

### Operating System

Debian 11

### n8n Version

1.29.1

### Node.js Version

18.17.0

### Database

SQLite (default)

### Execution mode

main (default)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This PR overhaul the code that updates `document.title`, and also updates most of the views to set relevant titles.

## Related Linear tickets, Github issues, and Community forum posts

Fixes #8861
[CAT-78](https://linear.app/n8n/issue/CAT-78)

## Review / Merge checklist

- [x] PR title and summary are descriptive
- [x] Tests included

### CODE REVIEW

File: packages/editor-ui/src/composables/useDocumentTitle.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	useDefaultPrefix?: boolean;
+}
+
+const DEFAULT_TITLE = 'Workflow Automation';
+
+export function useDocumentTitle({ useDefaultPrefix }: Options = { useDefaultPrefix: true }) {
+	const settingsStore = useSettingsStore();
+	const { releaseChannel } = settingsStore.settings;
+	const defaultPrefix =
+		!releaseChannel || releaseChannel === 'stable' ? 'n8n' : `[${releaseChannel.toUpperCase()}]`;
```
Comment (alexgrozav) [inline packages/editor-ui/src/composables/useDocumentTitle.ts:13]: AFAIK, even when we add the release channel, `n8n` should still be part of the title.

```suggestion
		!releaseChannel || releaseChannel === 'stable' ? 'n8n' : `[${releaseChannel.toUpperCase()}] n8n`;
```

**Review by alexgrozav — CHANGES_REQUESTED** [review]
Looking good! One small remark regarding the release channel

**Review by alexgrozav — APPROVED** [review]
Thanks for addressing the suggestions!

**Comment by janober** [comment]
Got released with [`n8n@1.62.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.62.1)

**Pull request merged**